common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -86,30 +86,42 @@ def list(self, request, *args, **kwargs):
     def perform_destroy(self, instance):
         """
         This method is an override and is called by the DELETE method
+
+        This method will revoke and
         """
-        save_model = False
-        if instance.expired_at is None:
-            instance.expired_at = timezone.now()
-            log.info('Set expired_at to [%s] for course entitlement [%s]', instance.expired_at, instance.uuid)
-            save_model = True
-
-        if instance.enrollment_course_run is not None:
-            CourseEnrollment.unenroll(
-                user=instance.user,
-                course_id=instance.enrollment_course_run.course_id,
-                skip_refund=True
-            )
-            enrollment = instance.enrollment_course_run
-            instance.enrollment_course_run = None
-            save_model = True
-            log.info(
-                'Unenrolled user [%s] from course run [%s] as part of revocation of course entitlement [%s]',
-                instance.user.username,
-                enrollment.course_id,
-                instance.uuid
-            )
-        if save_model:
-            instance.save()
+        log.info(
+            'Entitlement Revoke requested for Course Entitlement[%s]',
+            str(instance.uuid)
+        )
+        process_revoke_and_unenroll_entitlement(instance)
+
+
+@transaction.atomic
+def process_revoke_and_unenroll_entitlement(course_entitlement):
+    save_model = False
+
+    if course_entitlement.expired_at is None:
+        course_entitlement.expired_at = timezone.now()
+        log.info('Set expired_at to [%s] for course entitlement [%s]', course_entitlement.expired_at, course_entitlement.uuid)
+        save_model = True
+
+    if course_entitlement.enrollment_course_run is not None:
+        CourseEnrollment.unenroll(
+            user=course_entitlement.user,
+            course_id=course_entitlement.enrollment_course_run.course_id,
+            skip_refund=True
+        )
+        enrollment = course_entitlement.enrollment_course_run
+        course_entitlement.enrollment_course_run = None
```

Comment:
Can we make use of the `_unenroll_entitlement` method defined below and DRY this up even more?